src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListIndexerProperty.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedReadOnlyListIndexerProperty : SynthesizedReadOnlyListPropertyBase
+    {
+        private readonly PropertySymbol _interfaceProperty;
+
+        internal SynthesizedReadOnlyListIndexerProperty(
+            SynthesizedReadOnlyListTypeSymbol containingType,
+            PropertySymbol interfaceProperty,
+            GenerateMethodBodyDelegate getAccessorBody,
+            GenerateMethodBodyDelegate? setAccessorBody) :
+            base(containingType)
+        {
+            Debug.Assert(setAccessorBody is null == interfaceProperty.SetMethod is null);
+
+            _interfaceProperty = interfaceProperty;
+            Name = ExplicitInterfaceHelpers.GetMemberName(interfaceProperty.Name, interfaceProperty.ContainingType, aliasQualifierOpt: null);
+            GetMethod = new SynthesizedReadOnlyListMethod(containingType, interfaceProperty.GetMethod, getAccessorBody);
+            SetMethod = interfaceProperty.SetMethod is null ? null : new SynthesizedReadOnlyListMethod(containingType, interfaceProperty.SetMethod, setAccessorBody!);
```

Comment:
todo; trying to figure out why we have teh set-method for a readonly list.

Reply:
The set method is for `IList<T>.this[]`, and the implementation throws a `NotSupportedException` in that case.